tests/bgp/test_bgp_suppress_fib.py

```diff
@@ -0,0 +1,585 @@
+    for test_item in traffic_data:
+        dst_ip = test_item[0]
+        expected_result = test_item[1]
+        ip_ver = ipaddress.ip_network(dst_ip.encode().decode(), False).version
+        logger.info("Testing with dst_ip = {} expected_result = {}"
+                    .format(dst_ip, expected_result))
+        pkt, exp_pkt = generate_packet(SRC_IP[ip_ver], dst_ip, router_mac)
+        tx_port = random.choice(ptf_interfaces)
+        if ptf_interfaces is recv_port:
+            if loop_back:
+                logger.info("Expected packet:\n dst_mac:{} - src_ip:{} - dst_ip:{} - ptf tx_port:{} - ptf rx_port:{}".
+                            format(router_mac, SRC_IP[ip_ver], dst_ip, tx_port, ptf_interfaces))
+                send_and_verify_loopback_packets(ptfadapter, pkt, exp_pkt, tx_port, ptf_interfaces, expected_result)
+            else:
+                logger.info("Loopback traffic - expected packet:\n dst_mac:{} - src_ip:{} - dst_ip:{} - ptf tx_port:{}\
+                 - ptf rx_port:{}".format(router_mac, SRC_IP[ip_ver], dst_ip, tx_port, tx_port))
+                send_and_verify_packet(ptfadapter, pkt, exp_pkt, tx_port, tx_port, expected_result)
+        else:
+            logger.info("Expected packet:\n dst_mac:{} - src_ip:{} - dst_ip:{} - ptf tx_port:{} - ptf rx_port:{}".
+                        format(router_mac, SRC_IP[ip_ver], dst_ip, tx_port, recv_port[ip_ver]))
+            send_and_verify_packet(ptfadapter, pkt, exp_pkt, tx_port, recv_port[ip_ver], expected_result)
+
+
+def announce_ipv4_ipv6_routes(ptf_ip, exabgp_port, exabgp_port_v6, action=ANNOUNCE):
+    """
+    Announce or withdraw ipv4 and ipv6 routes by exabgp
+    """
+    for route in IP_ROUTE_LIST:
+        announce_route(ptf_ip, route, exabgp_port, action)
+
+    for route in IPV6_ROUTE_LIST:
+        announce_route(ptf_ip, route, exabgp_port_v6, action)
+
+
+def config_bgp_suppress_fib(duthost, enable=True):
+    """
+    Enable or disable bgp suppress-fib-pending function
+    """
+    if enable:
+        logger.info('Enable BGP suppress fib pending function')
+        cmd = 'sudo config suppress-fib-pending enabled'
+    else:
+        logger.info('Disable BGP suppress fib pending function')
+        cmd = 'sudo config suppress-fib-pending disabled'
+    duthost.shell(cmd)
+
+
+def randomly_reboot(duthost, localhost):
+    """
+    Randomly choose operation from config reload/reboot/fast reboot/warm reboot
+    """
+    logger.info("Save configuration")
+    duthost.shell('sudo config save -y')
+
+    reboot_type_list = ["config_reload", "cold", "warm", "fast"]
+    reboot_type = random.choice(reboot_type_list)
+    logger.info("Randomly choose {} from config_reload, cold, warm, fast".format(reboot_type))
+    with allure.step("Do {}".format(reboot_type)):
+        if reboot_type == "config_reload":
+            config_reload(duthost)
+        else:
+            reboot(duthost, localhost, reboot_type=reboot_type, reboot_helper=None, reboot_kwargs=None)
```

Comment:
```suggestion
            reboot(duthost, localhost, reboot_type=reboot_type, reboot_helper=None, reboot_kwargs=None, wait_warmboot_finalizer=True)
```
To make sure warmboot is finished before you do the STOP/CONT on orchagent. Otherwise you interrup warm boot recovery and might not have the time needed for routes to re-apply.